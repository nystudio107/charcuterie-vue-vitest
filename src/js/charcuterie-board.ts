import {CharcuterieItemType} from "../@types/charcuterie-enums";

const charcuterieBoard: CharcuterieItem[] = [];

export function getAllItems() {
  return charcuterieBoard;
}

export function clearAllItems() {
  charcuterieBoard.length = 0;
}
export function getItemsOfType(type: CharcuterieItemType): CharcuterieItem[] {
  return charcuterieBoard.filter((value: CharcuterieItem) => value.type === type)
}

export function addItem(item: CharcuterieItem|undefined) {
  if (typeof item !== 'undefined') {
    charcuterieBoard.push(item);
  }
}
