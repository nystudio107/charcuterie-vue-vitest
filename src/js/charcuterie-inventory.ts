import {CharcuterieItemType} from '@/@types/charcuterie-enums';

const charcuterieInventory: CharcuterieItem[]  = [
  {
    name: 'salami',
    type: CharcuterieItemType.Meat,
    calories: 200,
  },
  {
    name: 'olives',
    type: CharcuterieItemType.Fruit,
    calories: 100
  }
];

export function getInventory() {
  return charcuterieInventory;
}

export function getInventoryItem(name: string): CharcuterieItem|undefined {
  return charcuterieInventory.find((value: CharcuterieItem) => value.name === name);
}
