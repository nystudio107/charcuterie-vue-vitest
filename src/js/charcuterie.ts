export enum CharcuterieItemType {
  Cheese,
  Meat,
  Fruit,
}

const charcuterieBoard: CharcuterieItem[] = [];

export function getAllItems() {
  return charcuterieBoard;
}

export function getItemsOfType(type: CharcuterieItemType) {
  return charcuterieBoard.filter((value:CharcuterieItem) => value.type === type)
}

export function addCharcuterieItem(item: CharcuterieItem) {
  charcuterieBoard.push(item);
}
