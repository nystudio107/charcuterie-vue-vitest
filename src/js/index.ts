import { createApp } from 'vue'
import '../css/style.css'
import App from '../vue/App.vue'

createApp(App).mount('#app')
