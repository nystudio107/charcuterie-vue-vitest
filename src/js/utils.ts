export const kebabCase = (string: string) => string;
