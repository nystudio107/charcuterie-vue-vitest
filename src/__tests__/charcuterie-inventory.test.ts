import {getAllItems, getInventoryItem} from '../js/charcuterie-inventory';
import {it, expect, expectTypeOf, describe} from 'vitest';

describe('charcuterie-inventory.ts', () => {
  it('should return some inventory items', () => {
    const items = getAllItems();
    expect(items.length).greaterThan(0);
  });

  it('should return an array of nothing but CharcuterieItems', () => {
    const items = getAllItems();
    items.forEach((item: CharcuterieItem) => {
      expectTypeOf(item).toEqualTypeOf<CharcuterieItem>();
    })
  });

  it('should have an "olives" item', () => {
    const olives = getInventoryItem('olives');
    expect(typeof olives !== 'undefined').toBe(true);
    expectTypeOf(olives).toEqualTypeOf<CharcuterieItem>();
    expect(olives.name).eq('olives');
  });
});
