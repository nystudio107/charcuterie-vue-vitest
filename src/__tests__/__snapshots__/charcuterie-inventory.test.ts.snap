// Vitest Snapshot v1

exports[`charcuterie-inventory.ts > should have all of the items from our spec 1`] = `
[
  {
    "name": "graham",
    "type": "cracker",
  },
  {
    "name": "cream",
    "type": "cracker",
  },
  {
    "name": "matzo",
    "type": "cracker",
  },
  {
    "name": "graham",
    "type": "cracker",
  },
  {
    "name": "oatcake",
    "type": "cracker",
  },
  {
    "name": "rice",
    "type": "cracker",
  },
  {
    "name": "salami",
    "type": "meat",
  },
  {
    "name": "soppressata",
    "type": "meat",
  },
  {
    "name": "calabrese",
    "type": "meat",
  },
  {
    "name": "mortadella",
    "type": "meat",
  },
  {
    "name": "prosciutto",
    "type": "meat",
  },
  {
    "name": "chorizo",
    "type": "meat",
  },
  {
    "name": "capicola",
    "type": "meat",
  },
  {
    "name": "cheddar",
    "type": "cheese",
  },
  {
    "name": "roquefort",
    "type": "cheese",
  },
  {
    "name": "swiss",
    "type": "cheese",
  },
  {
    "name": "pecorino-romano",
    "type": "cheese",
  },
  {
    "name": "parmigiano-reggiano",
    "type": "cheese",
  },
  {
    "name": "gruyere",
    "type": "cheese",
  },
  {
    "name": "provolone",
    "type": "cheese",
  },
  {
    "name": "peanuts",
    "type": "nut",
  },
  {
    "name": "almond",
    "type": "nut",
  },
  {
    "name": "cashew",
    "type": "nut",
  },
  {
    "name": "walnut",
    "type": "nut",
  },
  {
    "name": "brazil",
    "type": "nut",
  },
  {
    "name": "pistachio",
    "type": "nut",
  },
  {
    "name": "peanuts",
    "type": "nut",
  },
  {
    "name": "apple",
    "type": "fruit",
  },
  {
    "name": "olives",
    "type": "fruit",
  },
  {
    "name": "strawberry",
    "type": "fruit",
  },
  {
    "name": "cherry",
    "type": "fruit",
  },
  {
    "name": "blackberry",
    "type": "fruit",
  },
  {
    "name": "raspberry",
    "type": "fruit",
  },
  {
    "name": "pear",
    "type": "fruit",
  },
  {
    "name": "carrot",
    "type": "vegetable",
  },
]
`;
